packages/backend/src/plugins/search.ts

```diff
@@ -13,29 +13,55 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { useHotCleanup } from '@backstage/backend-common';
+import {
+  PluginDatabaseManager,
+  useHotCleanup,
+} from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import { DefaultCatalogCollator } from '@backstage/plugin-catalog-backend';
 import { createRouter } from '@backstage/plugin-search-backend';
+import { ElasticSearchSearchEngine } from '@backstage/plugin-search-backend-module-elasticsearch';
+import { PgSearchEngine } from '@backstage/plugin-search-backend-module-pg';
 import {
   IndexBuilder,
   LunrSearchEngine,
+  SearchEngine,
 } from '@backstage/plugin-search-backend-node';
-import { PluginEnvironment } from '../types';
-import { DefaultCatalogCollator } from '@backstage/plugin-catalog-backend';
 import { DefaultTechDocsCollator } from '@backstage/plugin-techdocs-backend';
-import { ElasticSearchSearchEngine } from '@backstage/plugin-search-backend-module-elasticsearch';
+import { Logger } from 'winston';
+import { PluginEnvironment } from '../types';
+
+async function createSearchEngine({
+  logger,
+  database,
+  config,
+}: {
+  logger: Logger;
+  database: PluginDatabaseManager;
+  config: Config;
+}): Promise<SearchEngine> {
+  if (config.has('search.elasticsearch')) {
+    return await ElasticSearchSearchEngine.fromConfig({
+      logger,
+      config,
+    });
+  }
+
+  if (await PgSearchEngine.supported(database)) {
+    return await PgSearchEngine.from({ database });
+  }
+
+  return new LunrSearchEngine({ logger });
+}
 
 export default async function createPlugin({
   logger,
   discovery,
   config,
+  database,
 }: PluginEnvironment) {
   // Initialize a connection to a search engine.
-  const searchEngine = config.has('search.elasticsearch')
-    ? await ElasticSearchSearchEngine.fromConfig({
-        logger,
-        config,
-      })
-    : new LunrSearchEngine({ logger });
+  const searchEngine = await createSearchEngine({ config, logger, database });
   const indexBuilder = new IndexBuilder({ logger, searchEngine });
 
   // Collators are responsible for gathering documents known to plugins. This
```

Comment:
@Fox32 I'm trying to use this for our backstage search, so, after initializing the connection, do we need to add collators? and can i write a collator to index api plugin?

Reply:
Hey, @Narayan-dasari for questions like this, you'll find the community on [discord](https://discord.com/invite/MUpMjP2).

Also check out the [search getting started guide](https://backstage.io/docs/features/search/getting-started) which is applicable to PG search as well.